modules/optimization/doc/content/modules/optimization/getting_started/InvOptTheory.md

```diff
@@ -0,0 +1,376 @@
+\begin{equation}\label{eq:objective}
+J(\mathbf{u}, \mathbf{p}) = \frac{1}{2}\sum_i \left(\mathbf{u}_i - \mathbf{\bar u}_i\right)^2 + \frac{\rho}{2}\sum_i \mathbf{p}^2 ,
+\end{equation}
+where the first integral is an L$_2$ norm or euclidean distance between the experimental data, $\mathbf{\bar u}$, and the simulated solution, $\mathbf{u}$.  The second integral provides Tikhonov regularization on the parameters, $\mathbf{p}$, for ill-posed problems where $\rho$ controls the amount of regularization.  Other types of regularization may also be used.
+
+Gradient-free optimization solvers only require a function to solve for the objective given in [eq:objective].  Solving for the objective only requires solving a forward problem to determine $\mathbf{u}$ and then plugging that into [eq:objective] to determine $J$.  The forward problem is defined as the FEM model of the experiment which the analyst should have already made before attempting to perform optimization.  The parameters that go into the forward problem (e.g. pressure distributions on sidesets or material properties) are adjusted by the optimization solver and the forward problem is recomputed.  This process continues until $J$ is below some user defined threshold. The basic gradient-free solver available in TAO is the simplex or Nelder-Mead method.  Gradient-free optimization solvers are robust and straight-forward to use.  Unfortunately, their computational cost scales exponentially with the number of parameters.  When the forward model is a computationally expensive FEM model, gradient-free approaches quickly become computationally expensive.
+
+Gradient-based optimization algorithms require fewer iterations but require functions to solve for the gradient vector and sometimes Hessians matrix.  TAO has `petsc_options` to evaluate finite difference based gradients and Hessians by solving the objective function multiple times with perturbed parameters, which also requires multiple solves of the forward problem.  Finite difference gradients and Hessians are good for testing an optimization approach but become too expensive for realistic problems.
+
+Given the large parameter space, we resort to the adjoint method for gradient computation; unlike finite difference approaches, the computational cost of adjoint methods is independent of the number of parameters [!citet](plessix2006review).  In the adjoint method, the gradient, i.e. the total derivative $\mathrm{d}J/\mathrm{d}\mathbf{p}$, is computed as,
+\begin{equation}\label{eq:adjointGrad}
+\frac{\mathrm{d}J}{\mathrm{d}\mathbf{p}} = \frac{\partial J}{\partial\mathbf{p}}+\mathbf{\lambda}\frac{\partial\mathbf{g}}{\partial\mathbf{p}} ,
```

Comment:
Not sure on this: maybe adding transpose superscript for \mathbf{\lambda}} can be clearer on indicating the final result is a vector in side of number of parameters.

Reply:
The eqn is right, eqn 12 in this paper clarifies the size of these vectors and matrices:  https://www.sciencedirect.com/science/article/pii/S0376042119300120
I added a citation to it.